File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = 0
        for i in range(len(nums) - 2):
            if nums[i] == 0:
                nums[i] = 1
                nums[i + 1] = abs(nums[i + 1] - 1) 
                nums[i + 2] = abs(nums[i + 2] - 1) 
                n += 1

        for num in nums:
            if num == 0:
                return -1

        return n

File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.java
class Solution {
    public int minOperations(int[] nums) {
        int n = 0;
        for (int i = 0; i < nums.length - 2; ++i)
            if (nums[i] == 0) 
            {
                ++n;
                nums[i] = 1;
                nums[i + 1] ^= 1;
                nums[i + 2] ^= 1;
            }
        return nums[nums.length - 1] == 1 && nums[nums.length - 2] == 1 ? n : -1;
    }
}